c#/Makefile

```diff
@@ -0,0 +1,10 @@
+all:
```

Comment:
This Makefile rule does not make sure that
  (a) the librgbmatrix.so is build. You need to depend on that
  (b) does not do anything depending on the change of the cs files. So you want to depend on each.
  (c) the test program is part of building the library, which is probably what you want to keep independent.

So you probably want to have something like this
```
CSHARP_LIB=RGBLedMatrix.dll
SOURCES=RGBLedCanvas.cs RGBLedMatrix.cs
CSHARP_COMPILER=mcs

RGB_LIBDIR=../lib
RGB_LIBRARY_NAME=rgbmatrix
RGB_LIBRARY=$(RGB_LIBDIR)/lib$(RGB_LIBRARY_NAME).so.1

$(CSHARP_LIB) : $(SOURCES) $(RGB_LIBRARY)
        $(CSHARP_COMPILER) -target:library -out:$@ $(SOURCES)

$(RGB_LIBRARY):
        $(MAKE) -C $(RGB_LIBDIR)
```

for the library, and then an independent target for the test program.